### ISSUE TITLE
support 5-letter language-codes (de-at, en-gb) correctly

### ISSUE THREAD

**Issue by nerdoc** [zyegfryed/django-statici18n#58]
When using 5-char `settings.LANGUAGE_CODE`s like "en-gb" or "de-at" (like me), after the setup of statici18n everything seems to work, but djangojs.js doesn't get loaded, `http://localhost:8000/static/jsi18n/de-at/djangojs.js` is a 404.
```
./manage.py compilejsi18n
processing language en
processing language de
```
After that command, in /static/jsi18n there are 2 directories: "en" and "de", so `jsi18n/de-at/djangojs.js` can't be found.

```django
<script type="application/javascript" src="{% statici18n LANGUAGE_CODE %}"></script>
{# is the same as
<script type="application/javascript" src="{% get_static_prefix %}jsi18n/{{ LANGUAGE_CODE }}/djangojs.js"></script>  #}

```
Maybe this is a misconfiguration on my side, but IMHO there should be automatically solved paths when using `{% statici18n %}`.

**Comment by nerdoc** [comment]
As additional info: the `compolejsi18n` management command does something:
```python
if locale is not None:
    languages = [locale]
elif not settings.USE_I18N:
    languages = [settings.LANGUAGE_CODE]
else:
    languages = [lang_code for (lang_code, lang_name) in settings.LANGUAGES]
```
in my case the third variable is used, and settings.LANGUAGES are taken as source, which are:
```python
LANGUAGES = [
    ("en", gettext("English")),
    ("de", gettext("German")),
]
```

So maybe this is the culprit, as LANGUAGES also should contain a language code as first part of the tuple. and if I set LANCUAGE_CODE to "de-at", I must use de-at in languages too...

If this is correct, this is my misconfiguration, and please close this issue... If not, I am thankful for every hint.

**Comment by zyegfryed** [comment]
Hi @nerdoc,

When relying on [`settings.LANGUAGE_CODE`](https://docs.djangoproject.com/en/5.0/ref/settings/#language-code), you can generate the Javascript Catalog by leveraging the `locale` option:

```shell
./manage.py compilejsi18n --locale <LANGUAGE_CODE>
```

TBH, I never configured a project with a `settings.LANGUAGE_CODE` which is not included in `settings.LANGUAGES`. May I ask you why you need such a configuration?

**Comment by nerdoc** [comment]
"need" is maybe the wrong word... Maybe it's just misconfig. But I am wondering why Django allows LANGUAGE_CODE being not one of LANGUAGES' keys.

Is it correct that it has to be there?

If so, why isn't Django doing something like:
```python
if LANGUAGE_CODE not in [e[0] for e in LANGUAGES]:
    raise ImproperlyConfigured(
        f"LANGUAGE_CODE must be one of: {', '.join([e[0] for e in LANGUAGES])}"
    )
```
in its setup? Should I propose that on their mailing list/forum?

**Comment by zyegfryed** [comment]
Django already provides the check [**translation.E004**](https://docs.djangoproject.com/en/5.0/ref/checks/#translation) to verify the [`settings.LANGUAGE_CODE` is part of the `settings.LANGUAGES`](https://github.com/django/django/blob/main/django/core/checks/translation.py#L58-L66).

This is actually a bug in `django-statici18n`! We will need to rely on [`django.utils.translation.get_supported_language_variant`](https://docs.djangoproject.com/en/5.0/ref/utils/#django.utils.translation.get_supported_language_variant) to coerce the provided LANGUAGE_CODE to the template tag.

Thanks for the catch!

**Comment by nerdoc** [comment]
oh, perfect, that's what I meant!

**Comment by zyegfryed** [comment]
Fixed in [v2.5.0](https://django-statici18n.readthedocs.io/en/v2.5.0/changelog.html#v2-5-0-2024-apr-20).
